numba/unicode.py

```diff
@@ -440,6 +453,80 @@ def endswith_impl(a, b):
         return endswith_impl
 
 
+@overload_method(types.UnicodeType, 'split')
+def unicode_split(a, sep, maxsplit=-1):
+    if isinstance(sep, types.UnicodeType):
```

Comment:
I think it is a requirement that `maxsplit` is an integer. The somewhat nonsensical:
```python
from numba import njit

def foo():
    return '1,2,3'.split(',', maxsplit=2.4)

print(njit(foo)())
```
currently compiles and returns `['1', '2', '3']`.

Reply:
Think that this is still an issue as of 1f0b6b6 ?